### PR TITLE
Add ticket code and dob to form

### PR DESCRIPTION
**Pull request by thecadams**
Hack this in for now, we will make these proper venue features when the venue access is implemented.

### CODE REVIEW

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:46]: [NB, nit] extract `CODE_CHECK_URL + data.code` into a semantically named variable

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:133]: Would this work?

```suggestion
          {errors?.date_of_birth?.type === "required" && (
```

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:131]: Here you only use `register`, but below you use `register({ require: true })`

Yet in the error message for `date_of_birth`, you're checking for a `required` error.

Is that correct? From reading the code, I would have expected this to also pass through the `required: true`, though i'm not familiar with the library semantics.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:157]: This block seems as though it will show all errors, not just `required`, but the above 1 line check seems like it won't necessarily display all errors? Is the one liner correct?

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:88]: [NB, nit] same as above RE: extracting semantically named variable

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:204]: This looks like a duplication of the code added in `src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx`? If so, can we extract a reusable 'dumb' component or two?

Eg. something like:

```jsx
<TicketCodeField register={register} error={errors?.date_of_birth} />
<DateOfBirthField register={register} error={errors?.code} />
```

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:86]: Is this onSubmit function the same between both forms? If so, can we extract it to a common function as well?

File: src/settings.ts
Comment (0xdevalias) [inline src/settings.ts:37]: [OOS, future] We really need to get around to making this a param/env var/similar

**Review by 0xdevalias — COMMENTED** [review]
A few comments/potential changes to DRY up the code a little.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (thecadams) [reply src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:131]: Just added required true, it's there on the code field so should be on dob too.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (thecadams) [reply src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:157]: It'll display all the errors because it'll default to the message unless the error is required. This code just makes the required error more informative and passes others through.

File: src/settings.ts
Comment (thecadams) [reply src/settings.ts:37]: Agreed 👍

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx
Comment (thecadams) [reply src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:86]: Good catch. There are a few key differences that mean it'd have to be progressively DRYed up:

* `signUp`/`signIn`
* append to `codes_used` (in `LogInForm`) vs set it for the first time (in `RegisterForm`)

OTOH the stuff that's the same includes

* Validation
* Post login steps

Would you be OK to hold off and DRY this up after I've removed the burn-related stuff from it?

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (thecadams) [reply src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:157]: In my recent changes, I've also pushed a change that unifies both form fields to behave the same way at this, since their behaviour appears to be intended to be identical. The DRY-up exposed the similarity.

File: src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx
Comment (0xdevalias) [reply src/components/organisms/AuthenticationModal/RegisterForm/RegisterForm.tsx:86]: Happy for it to be a future cleanup thing. I don't think it's worth blocking the other improvements here for it.

File: src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx
Comment (0xdevalias) [inline src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx:125]: We can use the new `DateOfBirthField` and `TicketCodeField` components instead of this now can't we?

**Review by 0xdevalias — APPROVED** [review]
[CRO] One spot of DRY missed I think, but otherwise this looks good to go. Will approve, but if you can clean that last one up as well it'd be 👌

**Pull request merged**
